test/exchangeIssuance/exchangeIssuance-staged.spec.ts

```diff
@@ -0,0 +1,906 @@
+        subjectTokenToApprove = setV2Setup.dai;
+      });
+
+      async function subject(): Promise<ContractTransaction> {
+        return await exchangeIssuance.approveToken(subjectTokenToApprove.address);
+      }
+
+      it("should update the approvals correctly", async () => {
+        const initUniswapDaiAllownace = await subjectTokenToApprove.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const initSushiswapDaiAllownace = await subjectTokenToApprove.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleDaiAllownace = await subjectTokenToApprove.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        await subject();
+
+        const finalUniswapDaiAllownace = await subjectTokenToApprove.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const finalSushiswapDaiAllownace = await subjectTokenToApprove.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const finalIssuanceModuleDaiAllownace = await subjectTokenToApprove.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        expect(finalUniswapDaiAllownace.sub(initUniswapDaiAllownace)).eq(MAX_UINT_96);
+        expect(finalSushiswapDaiAllownace.sub(initSushiswapDaiAllownace)).eq(MAX_UINT_96);
+        expect(finalIssuanceModuleDaiAllownace.sub(initIssuanceModuleDaiAllownace)).eq(MAX_UINT_96);
+      });
+    });
+
+    describe("#approveTokens", async () => {
+      let subjectTokensToApprove: StandardTokenMock[];
+
+      beforeEach(async () => {
+        subjectTokensToApprove = [setV2Setup.dai, setV2Setup.wbtc];
+      });
+
+      async function subject(): Promise<ContractTransaction> {
+        return await exchangeIssuance.approveTokens(subjectTokensToApprove.map(token => token.address));
+      }
+
+      it("should update the approvals correctly", async () => {
+        const token1 = subjectTokensToApprove[0];
+        const initUniswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const initSushiswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        const token2 = subjectTokensToApprove[1];
+        const initUniswapToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const initSushiswapToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        await subject();
+
+        const finalUniswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
```

Comment:
We typically like to compute the expected value here, so it would look something like:

```
const uniswapDaiAllowance = await dai.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
const expectedAllowance = initUniswapDaiAllowance.add(MAX_UINT_96);
expect(uniswapDaiAllowance).to.eq(expectedAllowance)
```